ricecooker/utils/downloader.py

```diff
@@ -41,3 +54,180 @@ def read(path, loadjs=False, session=None, driver=None):
     except (requests.exceptions.MissingSchema, requests.exceptions.InvalidSchema):
         with open(path, 'rb') as fobj:                          # If path is a local file path, try to open the file
             return fobj.read()
+
+
+def make_request(url, clear_cookies=True, timeout=60, *args, **kwargs):
+    sess = DOWNLOAD_SESSION
+
+    if clear_cookies:
+        sess.cookies.clear()
+
+    retry_count = 0
+    max_retries = 5
+    while True:
+        try:
+            response = sess.get(url, headers=headers, timeout=timeout, *args, **kwargs)
+            break
+        except (requests.exceptions.ConnectionError, requests.exceptions.ReadTimeout) as e:
+            retry_count += 1
+            print("Error with connection ('{msg}'); about to perform retry {count} of {trymax}."
+                  .format(msg=str(e), count=retry_count, trymax=max_retries))
+            time.sleep(retry_count * 1)
+            if retry_count >= max_retries:
+                return Dummy404ResponseObject(url=url)
```

Comment:
The `Dummy404ResponseObject` is missing.

It is defined here:  https://github.com/learningequality/sushi-chef-ck12/blob/master/chef.py#L77
but may be better to refactor to return `None` (or some other way to signal error).

Reply:
Changed to `raise e`